src/sentry/static/sentry/app/views/releases/releaseOverview.jsx

```diff
@@ -0,0 +1,184 @@
+
+import LoadingIndicator from '../../components/loadingIndicator';
+import LoadingError from '../../components/loadingError';
+import IssueList from '../../components/issueList';
+import FileChange from '../../components/fileChange';
+import CommitAuthorStats from '../../components/commitAuthorStats';
+import ReleaseProjectStatSparkline from '../../components/releaseProjectStatSparkline';
+import ApiMixin from '../../mixins/apiMixin';
+
+import {t} from '../../locale';
+
+
+const ReleaseOverview = React.createClass({
+  mixins: [ApiMixin],
+
+  getInitialState() {
+    return {
+      loading: true,
+      error: false,
+      projects: [],
+    };
+  },
+
+  componentDidMount() {
+    let {orgId, projectId, version} = this.props.params;
+
+    let path = `/projects/${orgId}/${projectId}/releases/${version}/commitfiles/`;
+    this.api.request(path, {
+      method: 'GET',
+      data: this.props.location.query,
+      success: (data, _, jqXHR) => {
+        this.setState({
+          error: false,
+          loading: false,
+          fileList: data,
+          pageLinks: jqXHR.getResponseHeader('Link')
+        });
+      },
+      error: () => {
+        this.setState({
+          error: true,
+          loading: false
+        });
+      }
+    });
+    this.getReleaseProjects();
+  },
+
+  getReleaseProjects() {
+    let {orgId, version} = this.props.params;
+    let path = `/organizations/${orgId}/releases/${version}/`;
+    this.api.request(path, {
+      method: 'GET',
+      success: (data, _, jqXHR) => {
+        this.setState({
+          projects: data.projects,
+          error: false,
+        });
+      },
+      error: () => {
+        this.setState({
+          error: true,
+        });
+      }
+    });
+  },
+
+  renderEmpty() {
+    return <div className="box empty">{t('No other projects affected.')}</div>;
+  },
+
+  render() {
+    let {orgId, projectId, version} = this.props.params;
+
+    if (this.state.loading)
+      return <LoadingIndicator/>;
+
+    if (this.state.error)
+      return <LoadingError/>;
+
+    let {fileList, projects} = this.state;
+
+    // convert list of individual file changes (can be
+    // multiple changes to a single file) into a per-file
+    // summary
+    let fileInfo = fileList.reduce(function (summary, fileChange) {
+      let {author, type, filename} = fileChange;
+      if (!summary.hasOwnProperty(filename)) {
+        summary[filename] = {
+          authors: {}, types: new Set()
+        };
+      }
+
+      summary[filename].authors[author.email] = author;
+      summary[filename].types.add(type);
+
+      return summary;
+    }, {});
+
+    let fileCount = Object.keys(fileInfo).length;
+    return (
+      <div>
+        <div className="row" style={{paddingTop: 10}}>
+          <div className="col-sm-8">
+            <h5>{fileCount} Files Changed</h5>
+            <ul className="list-group list-group-striped m-b-2">
+              {Object.keys(fileInfo).map(file => {
```

Comment:
So, we have underscore, which [has a map utility](http://underscorejs.org/#map) fn that will iterate over objects (for the future):

```javascript
{_.map(fileInfo, (desc, file) {
   // ...
}
```

Also – we should improve some of these variable names for clarity. Is `file` actually `filename`? Is `fileInfo` more like `fileChangeSummary`?